java/runtime/src/main/java/org/ray/runtime/task/ArgumentsBuilder.java

```diff
@@ -30,10 +33,15 @@ private static boolean checkSimpleValue(Object o) {
         data = Serializer.encode(arg);
       } else if (arg instanceof RayObject) {
         id = ((RayObject) arg).getId();
-      } else if (checkSimpleValue(arg)) {
-        data = Serializer.encode(arg);
-      } else {
+      } else if (arg instanceof byte[]) {
```

Comment:
Why do you have this case? Shouldn't it just go through the `else`?

Reply:
Yeah, it's not needed. Our internal code base needs this because for cross-language tasks, arguments (byte arrays) are always passed by reference. @bibabolynn please remove this.
btw, we'll start pushing code for cross-language support to github after #3028 is merged.